pkg/api/render.go

```diff
@@ -53,14 +54,19 @@ func (hs *HTTPServer) RenderToPng(c *contextmodel.ReqContext) {
 		headers["Accept-Language"] = acceptLanguageHeader
 	}
 
+	userID, errID := identity.UserIdentifier(c.SignedInUser.GetNamespacedID())
+	if errID != nil {
+		hs.log.Error("Failed to parse user id", "err", errID)
+	}
+
 	result, err := hs.RenderService.Render(c.Req.Context(), rendering.Opts{
 		TimeoutOpts: rendering.TimeoutOpts{
 			Timeout: time.Duration(timeout) * time.Second,
 		},
 		AuthOpts: rendering.AuthOpts{
 			OrgID:   c.SignedInUser.GetOrgID(),
-			UserID:  c.UserID,
-			OrgRole: c.OrgRole,
+			UserID:  userID,
```

Comment:
question: is there a danger in setting the `userID` to 0 here in case we have failed to `GetNamespacedID`?

Reply:
Probably not, the rendering service has been hitting it with user ID 0 already.
Still the type of endpoints I think we should revise in the future (stop it from being called with Anon for example)